### DISCUSSION TITLE
Saving the connection status into a file.

### DISCUSSION THREAD

**Discussion by SebaLedesma** [savonet/liquidsoap#2306] [User Support]
Hi:

I'm using liquidsoap 1.4.1.
I'm using output.icecast to streaming up to the server and it seems to work pretty well.
Now I'm trying create a script to save the connection status in a text file.

```
#!/usr/bin/liquidsoap

def updateStatus(estado) 
  file.write(data=estado, "/tmp/upstreaming.txt")
end

output.icecast(%mp3(bitrate=320), host = "radios.streamingdha.com.ar", port = 8080, mount="/stream", name= "Test Stream", password = "hackme", on_connect=updateStatus("Connected!"), input.alsa())
```

But I'm getting this error:
```
Error 5: this value has type
  bool (inferred at line 4, char 2-49)
but it should be a subtype of
  () -> _
```
How can I declare the return type to be '_' (aka 'void' for c/c++ programmers) ?

Thanks in advance
Sebas

**Comment by smimram** [comment]
I guess that `estado` is a boolean, which you need to convert to a string before writing to a file. So,
```
def updateStatus(estado) 
  file.write(data=string_of(estado), "/tmp/upstreaming.txt")
end
```

**Comment by SebaLedesma** [comment]
Thanks, I've tried your advice but the same error message appears.
As I see it , the return of myfunction 'updateStatus' it's deduced by the return of file.write,
and I need to declare it as 'void function'  in C semantics or  'procedure' in pascal semantics.
I'm not familiar with ML, so I'm trying to find some examples of how it works.

**Reply by rrolla** [reply]
Hey @SebaLedesma Liquidsoap function automatically return the last line in function as in most of the languages there is return keyword at end of function here Is a bit more: https://github.com/savonet/liquidsoap/discussions/2102

So i guess it should be like this
```ruby
def updateStatus(estado) 
  file.write(data=string_of(estado), "/tmp/upstreaming.txt")
  {}
end
```

**Comment by SebaLedesma** [comment]
Thanks!
With you suggestion the code it's now:
```
#!/usr/bin/liquidsoap

def updateStatus(estado) 
  file.write(data=string_of(estado), "/tmp/upstreaming.txt")
  ()
end

output.icecast(%mp3(bitrate=320), host = "radios.streamingdha.com.ar", port = 8080, mount="/stream", name= "Test Stream", password = "hackme", on_connect=updateStatus("Connected!"), input.alsa())
```
now I'm getting:
```
At line 4, char 2-49:
Warning 3: This expression should have type unit.
No output defined, nothing to do
```

**Reply by rrolla** [reply]
it should be `{}` Curly brackets not `()` Parentheses.

**Comment by SebaLedesma** [comment]
This works :)
```
def updateStatus(estado)
    ignore(file.write(data=estado, "/tmp/upstream.txt"))
end

```
